File: BluetoothLightLib/src/main/java/com/telink/util/Strings.java
/*
 * Copyright (C) 2015 The Telink Bluetooth Light Project
 *
 */
package com.telink.util;

import java.nio.charset.Charset;

public final class Strings {

    private Strings() {
    }

    public static byte[] stringToBytes(String str, int length) {

        byte[] srcBytes;

        if (length <= 0) {
            return str.getBytes(Charset.defaultCharset());
        }

        byte[] result = new byte[length];

        srcBytes = str.getBytes(Charset.defaultCharset());

        if (srcBytes.length <= length) {
            System.arraycopy(srcBytes, 0, result, 0, srcBytes.length);
        } else {
            System.arraycopy(srcBytes, 0, result, 0, length);
        }

        return result;
    }

    public static byte[] stringToBytes(String str) {
        return stringToBytes(str, 0);
    }

    public static String bytesToString(byte[] data) {
        return data == null || data.length <= 0 ? null : new String(data, Charset.defaultCharset()).trim();
    }

    public static boolean isEmpty(String str) {
        return str == null || str.trim().isEmpty();
    }
}

File: BluetoothLightLib/src/main/java/com/telink/util/Strings.kt
package com.telink.util

import java.nio.charset.Charset

object Strings {

    @JvmOverloads
    fun stringToBytes(str: String, length: Int = 0): ByteArray {

        val srcBytes: ByteArray = str.toByteArray(Charset.defaultCharset())

        if (length <= 0) {
            return str.toByteArray(Charset.defaultCharset())
        }
        val result = ByteArray(length)

        if (srcBytes.size <= length) {
            System.arraycopy(srcBytes, 0, result, 0, srcBytes.size)
        } else {
            System.arraycopy(srcBytes, 0, result, 0, length)
        }

        return result
    }

    fun bytesToString(data: ByteArray?): String? {
        return if (data == null || data.isEmpty()) null else String(data, Charset.defaultCharset()).trim { it <= ' ' }
    }

    fun isEmpty(str: String?): Boolean {
        return str == null || str.trim { it <= ' ' }.isEmpty()
    }
}
